docs/behat/basic-usage.rst

```diff
@@ -0,0 +1,510 @@
+         */
+        public function isSingleResourceOnPage(array $parameters)
+        {
+            try {
+                $rows = $this->tableAccessor->getRowsWithFields($this->getElement('table'), $parameters);
+                // Table accessor is a helper service which is responsible for all html table operations
+
+                return 1 === count($rows);
+            } catch (\InvalidArgumentException $exception) {
+                return false;
+            } catch (ElementNotFoundException $exception) {
+                return false;
+            }
+        }
+    }
+
+.. warning::
+
+    There is one small gap in this concept PageObjects aren't the concrete instance of current opened page they only mimic this behaviour (dummy pages).
+    This gap will be more understandable on code example.
+
+.. code-block:: php
+
+    // Of course this is only to illustrate this gap.
+
+    final class GapContext implements Context
+    {
+        private $homePage;
+        private $leagueIndexPage;
+
+        /**
+         * @Given I want to be on home page
+         */
+        public function iWantToBeOnHomePage() // After this method call we will be on "http://your_domain.com".
+        {
+            $this->homePage->open(); //When we add @javascript tag we can actually see this thanks to selenium.
+        }
+
+        /**
+         * @Then I want to read side bar and get The weather in France information
+         */
+        public function iWantToReadSideBarOnHomePage($someInformation) // Still "http://your_domain.com".
+        {
+            $someInformation === $this->leagueIndexPage->readMatchResults() // This returns true, but wait a second we are on home page (dummy pages).
+
+            $someInformation === $this->homePage->readWeather() // This also returns true.
+        }
+    }
+
+    class HomePage
+    {
+        // In this context on home page sidebar you have for example weather information in selected countries.
+        public function readWeather()
+        {
+            return $this->getElement('sidebar')->getText();
+        }
+
+        protected function getDefinedElements()
+        {
+            return ['sidebar' => ['css' => '.sidebar']]
+        }
+
+        protected function getUrl()
+        {
+            return 'http://your_domain.com';
+        }
+    }
+
+    class LeagueIndexPage
+    {
+        // In this context you have for example football match results.
+        public function readMatchResults()
+        {
+            return $this->getElement('sidebar')->getText();
+        }
+
+        protected function getDefinedElements()
+        {
+            return ['sidebar' => ['css' => '.sidebar']]
+        }
+
+        protected function getUrl()
+        {
+            return 'http://your_domain.com/leagues/'
+        }
+    }
+
+Registering context
+-------------------
+Like you see in previous section we have registered a lot of contexts so we will show you only some of step implementation.
```

Comment:
`As it was shown in the previous section we have registered a lot of contexts, so we will show you only some of the step implementation.`